### PR TITLE
[Auto-Parallel] Reshard API & Hybrid Parallel Unitest for dy2static mode

### PR DESCRIPTION
**Pull request by JZ-LIANG**
### PR types
Function optimization

### PR changes
Others

### Description

Pcard-76459

Reshard API in dy2static mode
MP-SP-DP-PP Hybrid Parallelism in dy2static mode

### CODE REVIEW

File: test/auto_parallel/hybrid_strategy/test_semi_auto_parallel_hybrid_strategy.py [hunk 1]
```diff
@@ -164,7 +164,7 @@ def test_simple_net_hybrid_strategy(self):
 class TestSemiAutoParallelLlama3D(test_base.CommunicationTestDistBase):
     def setUp(self):
         super().setUp(num_of_devices=8, timeout=200, nnode=1)
-        self._default_envs = {"dp": "2", "mp": "2", "pp": "2", "acc_step": "2"}
```
Comment (GhostScreaming) [inline test/auto_parallel/hybrid_strategy/test_semi_auto_parallel_hybrid_strategy.py:167]: 现在acc_step=2是会挂吗

File: test/auto_parallel/hybrid_strategy/semi_auto_parallel_llama_model.py [hunk 2]
```diff
@@ -501,7 +501,7 @@ def _prepare_decoder_attention_mask(
                     combined_attention_mask = dist.shard_tensor(
                         combined_attention_mask,
                         get_mesh(),
-                        [dist.Shard(0), dist.Replicate()],
+                        [dist.Replicate(), dist.Replicate()],
```
Comment (zhiqiu) [inline test/auto_parallel/hybrid_strategy/semi_auto_parallel_llama_model.py:504]: why change to  Replicate?

File: test/auto_parallel/hybrid_strategy/test_semi_auto_parallel_hybrid_strategy.py (hunk 1 above)
Comment (zhiqiu) [reply test/auto_parallel/hybrid_strategy/test_semi_auto_parallel_hybrid_strategy.py:167]: acc_step 需要strategy支持

**Review by zhiqiu — APPROVED** [review]
LGTM

**Review by tianshuo78520a — APPROVED** [review]

**Review by XieYunshen — APPROVED** [review]
LGTM
单测超时时间设置

**Review by zhiqiu — APPROVED** [review]
LGTM

**Review by JiabinYang — APPROVED** [review]
LGTM for _c_ops

**Pull request merged**